### DISCUSSION TITLE
Supported Filesystems

### DISCUSSION THREAD

**Discussion by 8thHalfHour** [ish-app/ish#1328] [General]
I have been unable to locate any information on the filesystems that are supported in iSH (distros, formats, etc) or their intended purpose. I was following around with them enough to know that it  seems to only like Linux `tar.gz` formats, but there appears to be more to it than that. For instance, it would not load a manually generated tar.gz of the alpine-extended ISO or the standard ISO. It did take the official Alpine minirootfs tar.gz, but it would not take the official Alpine netboot tar.gz. At least that was on my iPhone7 Plus; on my iPad Air 2, not even the official minirootfs loads as a bootable image.

So, I suppose I’m hoping to get some clarity here and find out what the filesystem option is intended to do (like what the original intention of it was), as well as what file formats it accepts and it’s limitations (ex., why did some Alpine variants work to load from while others didn’t, AND why so some seem to work only on certain devices?), and whether or not it would support other distros (tiny like puppy or full-featured distros like Debian or Arch).

(If this information does already exist somewhere, please just refer to it here; I certainly am not looking for you to have to duplicate it. I’m just looking for the information so I can mess around with and beta it properly, including that feature, having some idea of what it’s supposed to do and how it’s supposed to be able to do it. :D Appreciate anything you can offer on this subject. Thank you, in advance.)

**Comment by tbodt** [comment]
The tarball is imported as a Linux filesystem, and it will try to execute /sbin/init and /bin/login. So those need to exist and be compatible with iSH. This is the format of the alpine minirootfs tarballs. It's also the format you get when you export a filesystem from iSH.
